tests/unit/modules/test_virt.py

```diff
@@ -1896,6 +1896,49 @@ def test_update(self):
         with self.assertRaises(SaltInvocationError):
             virt.update("my_vm", boot={"efi": "Not a boolean value"})
 
+        # Update memtune parameter case
+        memtune = {
+            "soft_limit": "0.5g",
+            "hard_limit": "1024",
+            "swap_hard_limit": "2048m",
+            "min_guarantee": "1 g",
+        }
+
+        invalid_number = {
+            "soft_limit": "3.4.MB",
+        }
+
+        invalid_unit = {"soft_limit": "2HB"}
+
+        self.assertEqual(
+            {
+                "definition": True,
+                "disk": {"attached": [], "detached": [], "updated": []},
+                "interface": {"attached": [], "detached": []},
+            },
+            virt.update("my_vm", memtune=memtune),
+        )
+
+        setxml = ET.fromstring(define_mock.call_args[0][0])
+        self.assertEqual(
+            setxml.find("memtune").find("soft_limit").text, str(int(0.5 * 1024 ** 3))
+        )
+        self.assertEqual(
+            setxml.find("memtune").find("hard_limit").text, str(1024 * 1024 ** 2)
+        )
+        self.assertEqual(
+            setxml.find("memtune").find("swap_hard_limit").text, str(2048 * 1024 ** 2)
+        )
+        self.assertEqual(
+            setxml.find("memtune").find("min_guarantee").text, str(1 * 1024 ** 3)
+        )
+
+        with self.assertRaises(SaltInvocationError):
+            virt.update("my_vm", boot=invalid_unit)
+
+        with self.assertRaises(SaltInvocationError):
+            virt.update("my_vm", boot=invalid_number)
```

Comment:
Aren't those covered by the `_handle_unit()` test normally? If so there is no need to add them here too IMHO.

Reply:
You're right. we don't need these lines.

Reply:
oh sure it doesn't harm... you can keep them around for sure. No problem to have more tests than the minimum